### PR TITLE
Konnectd embedd

### PR DESCRIPTION
**Pull request by IljaN**
Note that kopano frontend is currently committed, and is copied via make in to the bin directory.
Will work on embedding in a seperate Task (#8 )

Konnect is pulled from https://github.com/IljaN/konnect as upstream needs more time for reviewing https://github.com/Kopano-dev/konnect/pull/19

Set env vars:
```
LDAP_URI=ldap://localhost:389
LDAP_BINDDN=Administrator@samdom
LDAP_BINDPW=Password1!
LDAP_BASEDN=DC\=samdom,DC\=example,DC\=com
LDAP_SCOPE=sub
LDAP_LOGIN_ATTRIBUTE=mail
LDAP_EMAIL_ATTRIBUTE=mail
LDAP_NAME_ATTRIBUTE=cn
LDAP_UUID_ATTRIBUTE=sAMAccountName
LDAP_UUID_ATTRIBUTE_TYPE=text
LDAP_FILTER=(objectClass\=organizationalPerson)
KOPANO_DEBUG_SERVER_REQUEST_LOG=1
KONNECTD_IDENTITY_MANAGER=ldap
KONNECTD_ISS=https://127.0.0.1:9130
PHOENIX_OIDC_METADATA_URL=https://127.0.0.1:9130/.well-known/openid-configuration
PHOENIX_OIDC_AUTHORITY=https://127.0.0.1:9130
KONNECTD_IDENTIFIER_REGISTRATION_CONF=./identifier-registration.yaml
REVA_OIDC_ISSUER=https://127.0.0.1:9130
REVA_OIDC_INSECURE=1
REVA_USER_DRIVER=ldap
PHOENIX_WEB_CONFIG_SERVER=http://localhost:9140
KONNECTD_TRANSPORT_TLS_CERT=/home/ilja/code/ocis/server.crt
KONNECTD_TRANSPORT_TLS_KEY=/home/ilja/code/ocis/server.key
```
Generate certs for transport-encryption
```
openssl genrsa -out server.key 2048
openssl ecparam -genkey -name secp384r1 -out server.key
openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650
```
Run
```
ocis-konnectd --log-level debug server
```

Navigate to http://127.0.0.1:9130/signin/v1/identifier

Todo:
- [x] @tboerger konnectd builds a web ui. we should probably check it in, similar to the assets in ocis-hello / ocis-phoenix, correct? => Commited compiled version and added to make for now
- [ ] since we plan to build ocis simple with ocis-konnectd and ocis-devldap should we set the defaults to work with ocis-devldap? we can do that in ocis anyway. I would prefer if we set the defaults in ocis-konnectd to use an ldap server on the default port 389 ... or do we want to default to the devldap ports so that it will work out of the box even when starting the services independently (on the same machine) opinions?
- [x] upstream work to get https://github.com/Kopano-dev/konnect/pull/19 merged and get rid of the replace ... or reference your commit via `replace stash.kopano.io/kc/konnect v0.28.0 => github.com/IljaN/konnect v0.28.1-0.20200110083759-c7d0c1cc7dae`
- [x] fully wrap logger (WithFields, WithErrors does not work currently)

### CODE REVIEW

**Comment by butonic** [comment]
nice! I have this working with devldap using
```
LDAP_URI=ldap://localhost:9125 \
LDAP_BINDDN='cn=admin,dc=example,dc=org' \
LDAP_BINDPW=admin \     
LDAP_BASEDN='ou=users,dc=example,dc=org' \
LDAP_SCOPE=sub \
LDAP_LOGIN_ATTRIBUTE=uid \ 
LDAP_EMAIL_ATTRIBUTE=mail \ 
LDAP_NAME_ATTRIBUTE=cn \
LDAP_UUID_ATTRIBUTE=customuid \     
LDAP_UUID_ATTRIBUTE_TYPE=text \
LDAP_FILTER='(objectClass=person)' \
KOPANO_DEBUG_SERVER_REQUEST_LOG=1 \
./ocis-konnectd --log-level debug server --iss=https://127.0.0.1 ldap
```

**Comment by DeepDiver1975** [comment]
>     * since we plan to build ocis simple with ocis-konnectd and ocis-devldap should we set the defaults to work with ocis-devldap? we can do that in ocis anyway. I would prefer if we set the defaults in ocis-konnectd to use an ldap server on the default port 389 ... or do we want to default to the devldap ports so that it will work out of the box even when starting the services independently (on the same machine) opinions?

we wanted to use glauth as the one and only ldap server within ocis (it can proxy to real installations) - so defaulting to this one ocis-ldap (?) seems pretty reasonable - right?

File: go.mod [hunk 1]
```diff
@@ -6,14 +6,18 @@ require (
...
 	github.com/owncloud/ocis-pkg v1.2.0
+	github.com/rs/zerolog v1.17.2
+	github.com/sirupsen/logrus v1.4.2
 	github.com/spf13/viper v1.6.1
+	github.com/stretchr/testify v1.4.0
 	go.opencensus.io v0.22.2
+	stash.kopano.io/kc/konnect v0.28.0
 )
+
+replace stash.kopano.io/kc/konnect v0.28.0 => /home/ilja/code/konnect
```
Comment (refs) [inline go.mod]: you might want to remove this

Comment (butonic) [reply go.mod]: or replace it with his commit: `replace stash.kopano.io/kc/konnect v0.28.0 => github.com/IljaN/konnect v0.28.1-0.20200110083759-c7d0c1cc7dae` until the PR is merged

Comment (refs) [reply go.mod]: indeed, adding the comment not to forget about it :D

Comment (IljaN) [reply go.mod]: Will remove it when ready to merge.

**Review by DeepDiver1975 — APPROVED** [review]

**Pull request merged**